[17.0][MIG] datev_export_xml: Migration to 17.0

victoralmau opened the pull request:
Migration to 17.0

Locked by:
- [x] `datev_export`: https://github.com/OCA/l10n-germany/pull/173

@Tecnativa TT54915

fkantelberg committed:
datev_export_xml: Add module to export account.move as DATEV XML

fkantelberg committed:
Move out common configurations to base module

Update datev_export_xml.pot

README.rst

icon.png

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-germany-14.0/l10n-germany-14.0-datev_export_xml
Translate-URL: https://translation.odoo-community.org/projects/l10n-germany-14-0/l10n-germany-14-0-datev_export_xml/

fkantelberg committed:
[MIG] datev_export_xml: Migration to 15.0

fkantelberg committed:
[FIX] datev_export_xml: Fix call of sorted for pickings

fkantelberg committed:
[FIX] datev_export_xml: Enforce some limits and use different field for tax amount

fkantelberg committed:
[IMP] datev_export_xml: Check invoices before rendering to prevent badly described exceptions

fkantelberg committed:
[FIX] datev_export_xml: Negate values for refunds (invoice corrections)

fkantelberg committed:
[FIX] Problem when tax_base_amount is 0 or line names fallback to False

fkantelberg committed:
[FIX] datev_export_xml: Skip non-product lines

[UPD] Update datev_export_xml.pot

[BOT] post-merge updates

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-germany-15.0/l10n-germany-15.0-datev_export_xml
Translate-URL: https://translation.odoo-community.org/projects/l10n-germany-15-0/l10n-germany-15-0-datev_export_xml/

Commit:
[IMP] : pre-commit stuff

Commit:
[MIG] : Migration datev_export_xml to 16.0

Commit:
[IMP] datev_export_xml: Update template invoice_party account

fkantelberg committed:
[IMP] datev_export_xml: Allow previous validation to see all errors on invoices. Fix some edge cases like fully discounted lines

fkantelberg committed:
[FIX] datev_export_xml: Add general check on invoice e.g. when multiple taxes exists

Commit:
[FIX] datev_export_xml: display smart buttons

[UPD] Update datev_export_xml.pot

[BOT] post-merge updates

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-germany-16.0/l10n-germany-16.0-datev_export_xml
Translate-URL: https://translation.odoo-community.org/projects/l10n-germany-16-0/l10n-germany-16-0-datev_export_xml/

victoralmau committed:
[MIG] datev_export_xml: Migration to 17.0

TT54915

david-fleischmann commented:
@victoralmau 
![image](https://github.com/user-attachments/assets/02737f81-7cb1-4257-819d-0b8cfcbd750e)
Something is not correct yet in that version. I'm unable to see errors in the xml export to fix them. In v14 there was a Validate button where I had a clear overview of the erros in the accoun.move records:
![image](https://github.com/user-attachments/assets/cf05acda-b116-4e0c-b674-93d6ee7f11fc)

david-fleischmann commented:
@victoralmau @fkantelberg 
Just found that the display of the exception_info seems not to work as in V16 (runbot):
![image](https://github.com/user-attachments/assets/30865697-91ab-4952-ac66-7acb715462be)

In V17 the rendering of the error message seems to fail:
![image](https://github.com/user-attachments/assets/2de3ae0f-cb9d-46fa-81a7-686645f6cdca)

victoralmau committed:
[FIX] datev_export_xml: Fix tests (create journals if they do not exist)

[UPD] Update datev_export_xml.pot

[BOT] post-merge updates

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-germany-16.0/l10n-germany-16.0-datev_export_xml
Translate-URL: https://translation.odoo-community.org/projects/l10n-germany-16-0/l10n-germany-16-0-datev_export_xml/

victoralmau commented:
> @victoralmau ![image](https://private-user-images.githubusercontent.com/73656049/430325061-02737f81-7cb1-4257-819d-0b8cfcbd750e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDUzMTYwMjAsIm5iZiI6MTc0NTMxNTcyMCwicGF0aCI6Ii83MzY1NjA0OS80MzAzMjUwNjEtMDI3MzdmODEtN2NiMS00MjU3LTgxOWQtMGI4Y2ZjYmQ3NTBlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDIyVDA5NTUyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU4MGJjODJhZDBmOTE1M2Y0ZGI4MTM1MjgwYmI4M2NhMmYyNGU5OTFmMzdmYzgzYmM3MzUzNDc2Y2VhYWQ4NTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.y1GcDomdukoqhqXS17EHYoCR4B6NWx0YIFt8g3j8hF4) Something is not correct yet in that version. I'm unable to see errors in the xml export to fix them. In v14 there was a Validate button where I had a clear overview of the erros in the accoun.move records: ![image](https://private-user-images.githubusercontent.com/73656049/430325675-cf05acda-b116-4e0c-b674-93d6ee7f11fc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDUzMTYwMjAsIm5iZiI6MTc0NTMxNTcyMCwicGF0aCI6Ii83MzY1NjA0OS80MzAzMjU2NzUtY2YwNWFjZGEtYjExNi00ZTBjLWI2NzQtOTNkNmVlN2YxMWZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDIyVDA5NTUyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY0NjdiYjE5NjYyYmI2YTBiNzhjYzg4MTExYWIzYzM4Mzc1MGQxYjQwNzA5MGI4ZmYzMGI1MTQyNGI2MDIxOTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.oUvBATkc2IWFUPC56ZFdn96wDMaNR10IWzNcPNzQCps)

You are right, actually this was added in v15 but not in v16, therefore, I have added it first in v16 https://github.com/OCA/l10n-germany/pull/179 and when it is merged, I will add it here.

IMO the comment https://github.com/OCA/l10n-germany/pull/174#issuecomment-2781614407 is related to the other change.

victoralmau committed:
[IMP] datev_export_xml: pre-commit auto fixes

pedrobaeza commented:
/ocabot migration datev_export_xml

Please remove the temporary commit.

etobella commented on datev_export_xml/tests/test_datev_export.py:
```diff
@@ -0,0 +1,868 @@
+        cls.refund_date = cls.today - timedelta(days=55)
+        cls.start_date = cls.today - timedelta(days=34)
+        cls.end_date = cls.today - timedelta(days=32)
+        cls.InvoiceObj.with_context(force_delete=True).search([]).unlink()
```
This line is problematic if we have other modules installed. 

For example, if you install sale and try to pass the test, it will fail (no other modules installed). The problem comes from the fact that odoo auto installs account_audit_trail by default on `l10n_de`.

The way to solve it is to add 
``` python
        if "check_account_audit_trail" in cls.env["res.company"]._fields:
            cls.env["res.company"].search([]).write({"check_account_audit_trail": False})
```
Can you add it?

pedrobaeza commented:
@etobella @david-fleischmann is this OK for you then?

etobella reviewed (approved):
LGTM

pedrobaeza commented:
/ocabot merge nobump
